bloom-housing/bloom
# fix: fallback and refresh translations

# Pull Request Template

## Issue Overview

This PR addresses #4080 and #4081


## Description

This PR does two things:
1) Creates a fallback in case there is no translation for a string for any reason
2) Adds logic to the translation service to check if any updates have been made to the listing or the associated preferences and will fetch fresh translations if those updates were made after the most recent translation entry.

## How Can This Be Tested/Reviewed?

Since the second fix above should prevent the use of the fallback, testing requires a bit of commenting and uncommenting.

1. Reseed on core.
2. Find a listing with an online application and without preferences (ie. Blue Sky Apartments)
3. Click Apply online
4. Open the network tab while looking at the choose language page
5. Click "Empezar"
6. In the network tab you should see the listing id and clicking it with preview, the response.
7. Click back and you should be on the choose language page again
8. Comment out line 205 to 225 in translation service

9. On the partner's portal, go to that listing and add a preference.
10. Return to the choose langauge tab and click empezar again.
11. This time in the network tab, you'll notice both the text field and untranslatedText fields are filled out with english for the multiselect question options. Previously, the text field would be missing resulting in blank fields for the user and a blocked application.
12. You can go through the application to ensure the preferences questions show and do not block the user
13. Go back to the choose-language of the same listing.
14. Uncomment out line 205 to 225 in translation service
15. Click empezar and see that the options in the network response are translated in the text field.
16. Go back once again to the choose-language step
17. On the partner's portal, edit one of the preferences that is attached to the listing (ie change the option text)
18. Back on the public side, click empezar for the last time and in the network tab you'll see the updated translations

Feel free to ping me though if a walkthrough of this would be easier!

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
